### DISCUSSION TITLE
Incoherent scala-style typeclasses?

### DISCUSSION THREAD

**Discussion by pshirshov** [flix/flix#6321] [Q&A] [answered]
Could you give me a hint if the language supports incoherent typeclasses?

In Scala my imports affect implementations and I can swap typeclass instances during compile-time and even in run-time (by importing from a `val`). 

Is anything like that possible in Flix?

**Comment by magnus-madsen** [comment, marked as answer]
No-- by design we do not want incoherence. For example, you only get one `Eq` instance per type.

Moreover, we resolve type classes at compile-time which is an enabler for optimizations.

**Reply by magnus-madsen** [reply]
If you want to change behavior at runtime you will have to rely on higher-order functions (and algebraic effects and handlers when they arrive).
